mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -1519,13 +1676,44 @@ impl MarketCoinOps for TendermintCoin {
 
     fn wait_for_confirmations(
         &self,
-        _tx: &[u8],
+        tx_bytes: &[u8],
         _confirmations: u64,
         _requires_nota: bool,
-        _wait_until: u64,
-        _check_every: u64,
+        wait_until: u64,
+        check_every: u64,
     ) -> Box<dyn Future<Item = (), Error = String> + Send> {
-        let fut = async move { Ok(()) };
+        let tx_raw: TxRaw = try_fus!(Message::decode(tx_bytes));
+        let tx = TransactionEnum::CosmosTransaction(CosmosTransaction { data: tx_raw });
+        let mut tx_hash = hex::encode_upper(tx.tx_hash().0);
+        tx_hash.make_ascii_uppercase();
```

Comment:
Some overhead is happening here:
1. Transaction is decoded.
2. `tx_hash()` encodes it to a vector of bytes again, allocating more memory for the already available data.
3. `tx_hash()` returns `BytesJson`, which is wrapper over `Vec<u8>` - one more heap allocation.
4. `tx_hash` is already upper case, but `make_ascii_uppercase` is called for it.

It can be replaced with following:
```rust
// Just sanity check
let _tx_raw: TxRaw = try_fus!(Message::decode(tx_bytes));
let tx_hash = hex::encode_upper(sha256(tx_bytes));
```